heat/core/_operations.py

```diff
@@ -153,23 +196,43 @@ def __binary_op(
     if out is not None:
         sanitation.sanitize_out(out, output_shape, output_split, output_device)
 
-    # promoted_type = types.promote_types(t1.dtype, t2.dtype).torch_type()
     if t1.split is not None:
+        output_split = t1.split
+        # TODO: implement `dndarray.create_bulk_lshape_maps`
+        t1.create_lshape_map()
+        t2.create_lshape_map()
         if len(t1.lshape) > t1.split and t1.lshape[t1.split] == 0:
             result = t1.larray.type(promoted_type)
         else:
+            if (
+                t2.split is not None
+                and not (t2.lshape_map[:, t2.split] == t1.lshape_map[:, t1.split]).all()
+                and t2.lshape_map[:, t2.split].sum() == t1.lshape_map[:, t1.split].sum()
```

Comment:
The last condition is not necessary:
1) `output_shape` is already determined, so shapes are compatible
2) no (shape)broadcasting is happening in the split dimension because of lines 152 to 175
-> elements in split-dimension are already equal